src/editor/editor_brush_ui.cpp

```diff
@@ -0,0 +1,418 @@
+//             \/                  \/          \//_____/            \/
+//  ______________________                           ______________________
+//                        T H E   W A R   B E G I N S
+//         Stratagus - A free fantasy real time strategy game engine
+//
+/**@name editor_brush.h - Assistant for brushes in the editor. */
+//
+//      (c) Copyright 2023-2025 by Alyokhin
+//
+//      This program is free software; you can redistribute it and/or modify
+//      it under the terms of the GNU General Public License as published by
+//      the Free Software Foundation; only version 2 of the License.
+//
+//      This program is distributed in the hope that it will be useful,
+//      but WITHOUT ANY WARRANTY; without even the implied warranty of
+//      MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//      GNU General Public License for more details.
+//
+//      You should have received a copy of the GNU General Public License
+//      along with this program; if not, write to the Free Software
+//      Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
+//      02111-1307, USA.
+//
+
+//@{
+
+/*----------------------------------------------------------------------------
+--  Includes
+----------------------------------------------------------------------------*/
+
+
+#include "editor_brush_ui.h"
+
+#include "editor.h"
+
+
+/*----------------------------------------------------------------------------
+--  Defines
+----------------------------------------------------------------------------*/
+
+/*----------------------------------------------------------------------------
+--  Variables
+----------------------------------------------------------------------------*/
+
+/*----------------------------------------------------------------------------
+--  Functions
+----------------------------------------------------------------------------*/
+
+void CSetOfCtrls::show()
+{
+	for(auto &ctrl : hiddenControls) {
+		ctrl->setVisible(true);
+	}
+	hiddenControls.clear();
+}
+
+void CSetOfCtrls::hide()
+{
+	if (!hiddenControls.empty()) {
+		return;
+	}
+	for(auto &ctrl : controls) {
+		if (ctrl->isVisible()) {
+			ctrl->setVisible(false);
+			hiddenControls.push_back(ctrl);
+		}
+	}
+}
+
+Vec2i CSetOfCtrls::bottomRight() const
+{
+	int x = 0;
+	int y = 0;
+	for (const auto ctrl : controls) {
+		x = std::max(x, ctrl->getX() + ctrl->getWidth());
+		y = std::max(y, ctrl->getY() + ctrl->getHeight());
+	}
+	return Vec2i(x, y);
+}
+
+void CBrushControlsUI::reloadBrushes()
+{
+	const auto prevSelectedItem = brushesList->getElementAt(brushSelect->getSelected());
+	const auto newBrushesList = std::make_unique<StringListModel>(Editor.brushes.getBrushesNames());
+	brushesList.reset(newBrushesList.get());
+	brushSelect->setListModel(brushesList.get());
+
+	const std::string currentBrush = Editor.brushes.getCurrentBrush().getName();
+	if (const auto idx = brushesList->getIdxOfElement(currentBrush) != -1) {
```

Comment:
I think it is a bug, and you wanted:
```suggestion
	if (const auto idx = brushesList->getIdxOfElement(currentBrush); idx != -1) {
```